### PR TITLE
add generic repositories and services to product service API

### PR DESCRIPTION
**Pull request by Genresal**

### CODE REVIEW

File: Services/ProductService/IVCRM.BLL/Services/OrderService.cs [hunk 1]
```diff
@@ -9,22 +8,23 @@
...
+            IPublishEndpoint publishEndpoint) : base(orderRepository, mapper)
         {
             _orderRepository = orderRepository;
             _customerRepository = customerRepository;
-            _mapper = mapper;
             _publishEndpoint = publishEndpoint;
         }
 
-        public async Task<Order> Create(Order model)
+        public new async Task<Order> Create(Order model)
```
Comment (moltenessence) [inline Services/ProductService/IVCRM.BLL/Services/OrderService.cs:27]: may be override??

File: Services/ProductService/IVCRM.DAL/Repositories/BaseRepository.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
+﻿using IVCRM.DAL.Entities.Interfaces;
+using IVCRM.DAL.Infrastructure;
+using IVCRM.DAL.Repositories.Interfaces;
+using Microsoft.EntityFrameworkCore;
+
+namespace IVCRM.DAL.Repositories
+{
+    public class BaseRepository<TEntity> : IBaseRepository<TEntity> where TEntity : class, IEntity
```
Comment (moltenessence) [inline Services/ProductService/IVCRM.DAL/Repositories/BaseRepository.cs:8]: put empty lines between methods, please

**Review by moltenessence — APPROVED** [review]

**Review by Glipchik — APPROVED** [review]

File: Services/ProductService/IVCRM.BLL/Services/OrderService.cs (hunk 1 above)
Comment (Genresal) [reply Services/ProductService/IVCRM.BLL/Services/OrderService.cs]: fixed.

File: Services/ProductService/IVCRM.DAL/Repositories/BaseRepository.cs (hunk 2 above)
Comment (Genresal) [reply Services/ProductService/IVCRM.DAL/Repositories/BaseRepository.cs:8]: fixed.

**Pull request merged**